### PR TITLE
[MIRROR] Period Era Re-Enactment: Basic autopunctuation

### PR DESCRIPTION
**Pull request by Steals-The-PRs**
## About The Pull Request

This PR introduces basic automatic punctuation for the following circumstances on living mobs only, and **only for says** (/me or emotes that aren't special say-emotes are unaffected):

- If a chat message does not end in any kind of punctuation (like a period (.), a comma (,), a question mark (?), an exclamation mark (!) or a hyphen (-)), a period is automatically added to the end.
- Any kind of lowercase 'i' between word boundaries is automatically converted to uppercase.

**To be completely clear: this does not affect PDA's, asterisk emotes (\*emotes), full emotes, announcements or basically any other non-speaking form of communication.** It affects speech, so says and radio chatter *only*. PDAspeak enjoyers can continue to sling slang at one another without The Man moderating their grammar.

I've strategically chosen these two changes to produce what I believe to be most comfortable compromise between an easy chatting experience and having dialogue look clean, while also allowing for people to continue to flavour their character's speech patterns as appropriate.

There is *one* caveat: if you make use of in-game bold/italics/underline formatting, the autopunct function this PR adds will **not** add an additional end-of-line period if your message does not contain one. For example: `|I can't believe you've done this|` will not have a period added - you'll need to do this yourself.

I've confirmed with our unofficial TGcode liaison (goof) that regex filters have negligible performance impacts (and that any concerns that they did was largely what he describes as 'cargo cult' behavior). See for yourself on /tg/'s Discord [here](https://discord.com/channels/326822144233439242/326831214667235328/1186135072617799690), if you like.

tl;dr: easier and faster Good Chatting (tm) with likely little performance impact

## How This Contributes To The Nova Station Roleplay Experience

Dialogue becomes much cleaner and coincidentally, easier to type. Reach across your keyboards for the period key no more - chat as you would on Discord, and have the game prettify most of it for you. Never again suffer the tyranny of reaching for your shift-key mid conversation and adding four seconds onto your message typing time.

It is surprisingly comfortable. I encourage doubters to clone this locally and see for themselves.

Importantly, it makes interacting *easier*. If you're a slow typer trying to meet standards, this can legitimately shave off seconds per message - if you're sending hundreds of messages a round, you're responding more quickly and so are the people participating with you. It's a genuinely objective improvement in ways beyond just 'chat look good'.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>

https://github.com/Skyrat-SS13/Skyrat-tg/assets/966289/123f5f06-b76a-4e42-af8d-ebda8619e7b7

</details>

## Changelog

:cl:  yooriss
add: Talking now automatically capitalizes any lowercase 'i' in the middle of sentences, and ends sentences in a period if no other punctuation is present.
/:cl:

### CODE REVIEW

**Comment by AnywayFarus** [comment]
вот с i годно
